emoji-ios/src/main/java/com/vanniktech/emoji/ios/IosEmoji.java

```diff
@@ -58,24 +64,70 @@ public IosEmoji(final int codePoint, final int x, final int y, final boolean isD
 
   public IosEmoji(@NonNull final int[] codePoints, final int x, final int y, final boolean isDuplicate,
                      final Emoji... variants) {
-    super(codePoints, -1, isDuplicate, variants);
+    super(codePoints, -1, isDuplicate, false, variants);
+
+    this.x = x;
+    this.y = y;
+  }
+
+  public IosEmoji(@NonNull final int[] codePoints, final int x, final int y, final boolean isDuplicate,
+                  final boolean isAnimated, final Emoji... variants) {
+    super(codePoints, -1, isDuplicate, isAnimated, variants);
 
     this.x = x;
     this.y = y;
   }
 
   @Override @NonNull public Drawable getDrawable(final Context context) {
+
     final CacheKey key = new CacheKey(x, y);
     final Bitmap bitmap = BITMAP_CACHE.get(key);
-    if (bitmap != null) {
-      return new BitmapDrawable(context.getResources(), bitmap);
-    }
+
+    // TODO: Implement returning bitmapdrawable if existing based on one or several bitmaps existing
+    // if (bitmap != null) {
+    //  return new BitmapDrawable(context.getResources(), bitmap);
+    // }
+
+    return getDrawable(context, key);
+  }
+
+  private Drawable getDrawable(final Context context, CacheKey key) {
     final Bitmap strip = loadStrip(context);
     final Bitmap cut = Bitmap.createBitmap(strip, 1, y * SPRITE_SIZE_INC_BORDER + 1, SPRITE_SIZE, SPRITE_SIZE);
     BITMAP_CACHE.put(key, cut);
     return new BitmapDrawable(context.getResources(), cut);
   }
 
+  @Override @NonNull public AnimationDrawable getAnimatedDrawable(final Context context) {
+    Log.d("TAG", "Still working in AnimatedDrawable");
+
+    final Bitmap strip = loadStrip(context);
+    final Bitmap[] cut = loadAnimatedStrip(strip);
+
+    AnimationDrawable animationDrawable = new AnimationDrawable();
+    for (Bitmap bitmap : cut) {
+      animationDrawable.addFrame(new BitmapDrawable(context.getResources(), bitmap), 400);
+    }
+    animationDrawable.setOneShot(false);
+
+    System.out.println("Frames in getAnimatedDrawable Method: " + animationDrawable.getNumberOfFrames());
+
+    return animationDrawable;
+  }
+
+  private Bitmap[] loadAnimatedStrip(Bitmap bitmap) {
```

Comment:
I think https://developer.android.com/reference/android/graphics/drawable/AnimatedVectorDrawable or some gif implementation is a more performant solution instead of loading each bitmap individually.

Reply:
@rubengees I was thinking of using AnimatedImageDrawable: https://developer.android.com/reference/android/graphics/drawable/AnimatedImageDrawable
It allows for using GIF files directly using an ImageView. However, it requires API Version 28
which is fairly new, is this an option?